x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -612,6 +619,21 @@ protected Clock getClock() {
         return components;
     }
 
+    protected void possiblySetElasticPassword(SecurityIndexManager.State previousState, SecurityIndexManager.State currentState) {
+        if (previousState.equals(SecurityIndexManager.State.UNRECOVERED_STATE)
+            && currentState.equals(SecurityIndexManager.State.UNRECOVERED_STATE) == false
+            && securityIndex.get().indexExists() == false
```

Comment:
what should happen if security index gets deleted ? Would bootstrapping this again to set value be valuable/surprising/wrong?

Reply:
The idea around this is that we should be only setting this as the password of the elastic user, the first time the node starts after a package installation and only if the password of the elastic user hasn't been already set ( which would have triggered the auto-creation of the security index ) Since we are not triggering the creation of the security index during package installation and this would run early enough in the node lifecycle, there should be nothing unrelated that has auto-created the security index.

Reply:
> For this I think it is better to insert this action in the SecurityIndexManager#prepareIndexIfNeededThenExecute, right before calling the execute runnable argument. 

This is part of the alternative approach you are discussing in the other comment , right ?